app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java

```diff
@@ -601,12 +605,25 @@ protected void setupButtons(final StatusActionListener listener, final String ac
             }
         });
         if (reblogButton != null) {
-            reblogButton.setEventListener((button, buttonState) -> {
-                int position = getAdapterPosition();
-                if (position != RecyclerView.NO_POSITION) {
-                    listener.onReblog(buttonState, position);
-                }
-            });
+            if (statusDisplayOptions.confirmReblogs()) {
+                reblogButton.onClickListener = (buttonStatus) -> {
+                    int okButtonTextId =
+                            buttonStatus ? R.string.action_reblog : R.string.action_unreblog;
+                    new AlertDialog.Builder(reblogButton.getContext())
+                            .setMessage(statusContent)
```

Comment:
Shouldn't this be clamped to some maximum length? Or does the dialog do this automatically?

Reply:
I checked and dialog just becomes scrollable but it's a good point